src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -201,11 +202,14 @@ static void softKernelLoop(char *name, char *path, uint32_t cu_idx)
     }
 
     /* Reg file indicates the kernel should not be running. */
-    if (args_from_host[0] != 0x1)
-      continue;
+    if (!(args_from_host[0] & 0x1))
+      continue; //AP_START bit is not set; New Cmd is not available
 
     /* Start run the soft kernel. */
-    kernel(&args_from_host[1], &ops);
+    kernel_return = kernel(&args_from_host[1], &ops);
+    kernel_return &= 0xFF000000; //Keep only upper 8 bits
```

Comment:
I did not quite understand this "Keep only upper 8 bits". The signature of soft kernel function looks like
int soft_kernel(void *args, struct sk_operations *ops);
so it returns an integer. e.g. -EINVAL (-22, 0xFFFFFFEA). but after 
kernel_return &= 0xFF000000;
we got kernel_return = 0xFF00000000. We don't get any information from it, do we?

Reply:
changing it to save 32 bit int in regamp[1]
Host side change will be in separate PR